src/dune/predicate_lang.mli

```diff
@@ -2,6 +2,37 @@
 
 open! Stdune
 
+module Ast : sig
```

Comment:
I think it's probably simpler to drop this Ast submodule and to just have this type be in the main module. Instead, we can have specialized modules for integers or predicates.

It doesn't have to be done as part of this PR though.

Reply:
OK I leave it for later then :)